FlashCards.AhmadJer99/FlashCards/Data/StudyDBController.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Dapper;
+using FlashCards.Models;
+
+namespace FlashCards.Data;
+
+internal class StudyDBController : DataBaseController<StudySession>
+{
+    public StudyDBController()
+    {
+        InitDataBase();
+    }
+    public override void DeleteRow(int _id) // Not needed
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void InsertRow(StudySession studySession)
+    {
+        using (var connection = CreateConnection())
+        {
+            var insertQuery = "INSERT INTO study (FK_stack_id,session_date,score) VALUES (@FK_stack_id,@session_date,@score)";
+            connection.Execute(insertQuery, studySession);
+        }
+    }
+
+    public override List<StudySession> ReadAllRows()
+    {
+        using (var connection = CreateConnection())
+        {
+            var readQuery = "SELECT * FROM study";
+            List<StudySession> studySessions = connection.Query<StudySession>(readQuery).ToList();
+            return studySessions;
+        }
+    }public  List<StudySession> ReadAllRows(int stackId)
```

Comment:
🟠 **Code Style**

💡 You need to start caring about the code that you write. Adhere to the academy's [code conventions](https://thecsharpacademy.com/article/58/code-conventions).

🔧 Blank line between methods. Only one blank line between methods. Or if you have a preference that you want to stick with - be consistent. At the moment the blank lines are wildly inconsistent.

Reply:
Done fixed most of the violated coding conventions!